### PR TITLE
CLN: rolling step followups

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

xref https://github.com/pandas-dev/pandas/pull/45765

### CODE REVIEW

**Comment by jreback** [comment]
doc string check failing: https://github.com/pandas-dev/pandas/runs/5371271902?check_suite_focus=true

**Comment by jreback** [comment]
cc @rtpsw

File: pandas/tests/window/test_rolling.py [hunk 1]
```diff
@@ -96,71 +90,33 @@ def test_invalid_constructor(frame_or_series, w):
...
-
-@pytest.mark.parametrize(
-    "func",
-    [
-        lambda df: df.expanding,
-        lambda df: df.ewm,
-    ],
-)
-def test_constructor_step_unsupported(func, step):
+def test_step_not_implemented_for_cov_corr(agg):
```
Comment (jreback) [inline pandas/tests/window/test_rolling.py:111]: do we have another test which asserts NotImplemented for freq based step?

Comment (rtpsw) [reply pandas/tests/window/test_rolling.py:111]: `test_constructor_step_not_implemented` has the `window` parameter tested for "3D". Is this the freq test you're looking for?

Comment (mroeschke) [reply pandas/tests/window/test_rolling.py:111]: Yup, exactly (which was renamed `test_freq_window_not_implemented`)

File: pandas/tests/window/test_apply.py [hunk 2]
```diff
@@ -52,8 +52,7 @@ def test_rolling_apply_out_of_bounds(engine_and_raw):
 
 
 @pytest.mark.parametrize("window", [2, "2s"])
-@pytest.mark.parametrize("step", [None])
-def test_rolling_apply_with_pandas_objects(window, step):
+def test_rolling_apply_with_pandas_objects(window):
```
Comment (jreback) [inline pandas/tests/window/test_apply.py:55]: i would be ok with adding a step  with [None, 1] which should just work on the numeric window ones

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/window/test_apply.py (hunk 2 above)
Comment (jreback) [reply pandas/tests/window/test_apply.py:55]: great

**Review by jreback — APPROVED** [review]

**Pull request merged**